bookied/triggers/create.py

```diff
@@ -53,32 +59,48 @@ def createBmgs(self):
             uialist = list()
             uialist = bmg["asset"]
             name = bmg["description"]["en"]
-            x = 0 
+            x = 0
             while x < len(uialist):
-                bmg["asset"] =  uialist[x]
+                bmg["asset"] = uialist[x]
                 if(len(uialist) > 1):
-                   bmg["description"]["en"] = name + "_" + str(bmg["asset"]) 
-                   # To avoid duplicate BMG's on retriggering replays from DP's
-                   if "id" in bmg:
-                      bmg["id"] = None
-                 
+                    bmg["description"]["en"] = name + "_" + str(bmg["asset"])
+                    # To avoid duplicate BMG's on retriggering replays from DP's
+                    if "id" in bmg:
+                        bmg["id"] = None
+
                 x += 1
                 # Skip dynamic bmgs
                 if bmg["dynamic"]:
-                   # Dynamic BMGs are created separately
-                   log.debug("Skipping dynamic BMG: {}".format(str(bmg.identifier)))
-                   continue
+                    # Dynamic BMGs are created separately
+                    log.debug("Skipping dynamic BMG: {}".format(str(bmg.identifier)))
+                    continue
                 bmg.update()
-                self.createBms(bmg)
+                statusCreateBms = self.createBms(bmg)
+                if statusCreateBms == "midway":
+                    print("createBmms, toc > 10")
+                    return 'midway'
+
+            # Terminate loop and return midway if bmg creation took time.
+            # So that incident is processed later.
+            toc = time.time() - tic
+            if toc > 30:
```

Comment:
any criteria for selecting 30  secs over here, levee a comment

Reply:
If timeout is less than 30, a soccer event and all it's betting markets creation takes more than two attempts. With 30 or more number of attempts is 2, which is the lowest possible. Any attempt to get a soccer event created with single attempt consumes about 12 minutes if time out removed. Hence taking take into consideration soccer events, 30 seconds is an optimal timout for bmg creation.